Makefile

```diff
@@ -193,6 +198,30 @@ ENVTEST ?= $(LOCALBIN)/setup-envtest
 KUSTOMIZE_VERSION ?= v5.2.1
 CONTROLLER_TOOLS_VERSION ?= v0.15.0
 
+## Detect the platform (OS and architecture)
+OS := $(shell uname | tr '[:upper:]' '[:lower:]')
+ARCH := $(shell uname -m | sed -e 's/x86_64/amd64/' \
+                               -e 's/aarch64/arm64/' \
+                               -e 's/armv7l/arm/')
+
+.PHONY: kubectl
+kubectl: $(KUBECTL) ## Ensure kubectl exists, first check system path, then download it locally if necessary
+$(KUBECTL): $(LOCALBIN)
+	@if [ -n "$(SYSTEM_KUBECTL)" ]; then \
+		echo "Using system kubectl: $(SYSTEM_KUBECTL)"; \
+		ln -sf $(SYSTEM_KUBECTL) $(LOCALBIN)/kubectl; \
+	else \
+		echo "System kubectl not found, downloading to $(LOCALBIN)..."; \
+		if [ "$(OS)" = "darwin" ]; then \
+			curl -LO "https://dl.k8s.io/release/$$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/darwin/$(ARCH)/kubectl"; \
+		else \
+			curl -LO "https://dl.k8s.io/release/$$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/$(ARCH)/kubectl"; \
```

Comment:
We should probably use a fixed version here and just always use kubectl from LOCALBIN. The Makefile has the ENVTEST_K8S_VERSION version from the original operator-sdk setup, I believe, but the envtest bits have been removed. It might be a bit nicer codewise to use it, actually, as it will handle the arch for us. You can see an example in the plugin Makefile:
https://github.com/openshift-kni/oran-hwmgr-plugin/blob/main/Makefile#L120

It downloads the latest setup-envtest binary with the envtest target, which is then used to download kubectl and return its path.

For example:
```
dpenney ~/testing/envtest $ mkdir bin
dpenney ~/testing/envtest $ LOCALBIN=$PWD/bin
dpenney ~/testing/envtest $ GOBIN=$LOCALBIN go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
go: downloading sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20240927101401-4381fa0aeee4
dpenney ~/testing/envtest $ find bin
bin
bin/setup-envtest
dpenney ~/testing/envtest $ bin/setup-envtest use 1.28.3 --bin-dir $LOCALBIN -p path
/home/dpenney/testing/envtest/bin/k8s/1.28.3-linux-amd64

dpenney ~/testing/envtest $ find bin
bin
bin/setup-envtest
bin/k8s
bin/k8s/1.28.3-linux-amd64
bin/k8s/1.28.3-linux-amd64/etcd
bin/k8s/1.28.3-linux-amd64/kube-apiserver
bin/k8s/1.28.3-linux-amd64/kubectl
```

Reply:
The Makefile is updated.